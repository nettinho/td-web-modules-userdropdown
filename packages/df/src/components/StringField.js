import React from "react";
import PropTypes from "prop-types";
import { Form } from "semantic-ui-react";

export const StringField = ({ label, name, required, value, onChange }) => (
  <Form.Input
    label={label}
    required={required}
    value={value}
    name={name}
    onChange={onChange}
  />
);

StringField.propTypes = {
  label: PropTypes.string,
  name: PropTypes.string,
  required: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

export default StringField;
