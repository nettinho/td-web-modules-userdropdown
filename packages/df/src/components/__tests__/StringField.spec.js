import React from "react";
import { shallow } from "enzyme";
import { StringField } from "../StringField";

describe("<StringField />", () => {
  const label = "label";
  const name = "name";
  const required = true;
  const value = "Foo";
  const onChange = jest.fn();
  const props = { label, name, required, value, onChange };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<StringField {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
