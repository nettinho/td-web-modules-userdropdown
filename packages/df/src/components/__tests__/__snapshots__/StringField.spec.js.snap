// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StringField /> matches the latest snapshot 1`] = `
<FormInput
  as={[Function]}
  control={[Function]}
  label="label"
  name="name"
  onChange={[MockFunction]}
  required={true}
  value="Foo"
/>
`;
